src/modules/eventlisteners.ts

```diff
@@ -1,6 +1,12 @@
 import {VNode, VNodeData} from '../vnode';
 import {Module} from './module';
 
+export type On = {
+  [N in keyof HTMLElementEventMap]: (ev: HTMLElementEventMap[N]) => void
+} & {
+  [event: string]: EventListener
+};
```

Comment:
@paldepind is this fine? Notice that on line 6, while your example had `|` (union), I've found that `&` (intersection) works here.